src/v/coproc/pacemaker.cc

```diff
@@ -46,30 +50,44 @@ pacemaker::pacemaker(ss::socket_address addr, ss::sharded<storage::api>& api)
       wasm_transport_cfg(addr), wasm_transport_backoff()),
     api.local()) {}
 
+ss::future<> pacemaker::start() {
+    _offs.timer.set_callback([this] {
+        (void)ss::with_gate(_offs.gate, [this] {
+            return save_offsets(_offs.snap, _ntps).then([this] {
```

Comment:
what prevents the timer from scheduling concurrent snapshotting? would that be safe?

Reply:
Not sure, but off the top of my head, i wouldn't want that, I forgot to add some checks here before re-arming the timer, will do